docs/fundamentals/code-analysis/style-rules/language-rules.md

```diff
@@ -0,0 +1,1920 @@
+
+> [!TIP]
+>
+> Starting in Visual Studio 2019 version 16.3, you can configure code style rules from the [Quick Actions](/visualstudio/ide/quick-actions.md) light bulb menu after a style violation occurs. For more information, see [Automatically configure code styles in Visual Studio](/visualstudio/ide/editorconfig-language-conventions#automatically-configure-code-styles-in-visual-studio)
+
+## .NET style rules
+
+The style rules in this section are applicable to both C# and Visual Basic.
+
+- ["This." and "Me." qualifiers](#this-and-me)
+  - dotnet\_style\_qualification\_for_field
+  - dotnet\_style\_qualification\_for_property
+  - dotnet\_style\_qualification\_for_method
+  - dotnet\_style\_qualification\_for_event
+- [Language keywords instead of framework type names for type references](#language-keywords)
+  - dotnet\_style\_predefined\_type\_for\_locals\_parameters_members
+  - dotnet\_style\_predefined\_type\_for\_member_access
+- [Modifier preferences](#normalize-modifiers)
+  - dotnet\_style\_require\_accessibility_modifiers
+  - visual\_basic\_preferred\_modifier_order
+  - dotnet\_style\_readonly\_field
+- [Parentheses preferences](#parentheses-preferences)
+  - dotnet\_style\_parentheses\_in\_arithmetic\_binary\_operators
+  - dotnet\_style\_parentheses\_in\_other\_binary\_operators
+  - dotnet\_style\_parentheses\_in\_other\_operators
+  - dotnet\_style\_parentheses\_in\_relational\_binary\_operators
+- [Expression-level preferences](#expression-level-preferences)
+  - dotnet\_style\_object_initializer
+  - dotnet\_style\_collection_initializer
+  - dotnet\_style\_explicit\_tuple_names
+  - dotnet\_style\_prefer\_inferred\_tuple_names
+  - dotnet\_style\_prefer\_inferred\_anonymous\_type\_member_names
+  - dotnet\_style\_prefer\_auto\_properties
+  - dotnet\_style\_prefer\_conditional\_expression\_over\_assignment
+  - dotnet\_style\_prefer\_conditional\_expression\_over\_return
+  - dotnet\_style\_prefer\_compound\_assignment
+- ["Null" checking preferences](#null-checking-preferences)
+  - dotnet\_style\_coalesce_expression
+  - dotnet\_style\_null_propagation
+  - dotnet\_style\_prefer\_is\_null\_check\_over\_reference\_equality\_method
+- [Parameter preferences](#parameter-preferences)
+  - dotnet\_code\_quality\_unused\_parameters
+
+### <a name="this-and-me"></a>"This." and "Me." qualifiers
+
+This style rule can be applied to fields, properties, methods, or events. A value of **true** means prefer the code symbol to be prefaced with `this.` in C# or `Me.` in Visual Basic. A value of **false** means prefer the code element _not_ to be prefaced with `this.` or `Me.`.
+
+These rules could appear in an *.editorconfig* file as follows:
+
+```ini
+# CSharp and Visual Basic code style settings:
+[*.{cs,vb}]
+dotnet_style_qualification_for_field = false:suggestion
```

Comment:
If we're moving away from this syntax, we should replace it either in this PR or as the individual rule articles are added.

Reply:
Yes, I am hoping to take up https://github.com/dotnet/roslyn/issues/44201 next so severity suffix here is optional. Once that is done, I will update all these docs to not mention severity as part of this syntax. Users can still specify this in VS, but it will not be documented, and eventually deprecated once we have a full fledged UI for EditorConfig.